ts/private/ts_project_worker.js

```diff
@@ -0,0 +1,384 @@
+            return timestamp !== lastRequestTimestamp
+        }.bind(null, lastRequestTimestamp),
+        throwIfCancellationRequested: function (timestamp) {
+            if (timestamp !== lastRequestTimestamp) {
+                throw new ts.OperationCanceledException()
+            }
+        }.bind(null, lastRequestTimestamp),
+    }
+
+    timingStart("emit")
+    const result = program.emit(undefined, undefined, cancellationToken)
+    timingEnd("emit")
+
+    timingStart("diagnostics")
+    const diagnostics = ts.getPreEmitDiagnostics(
+        program,
+        undefined,
+        cancellationToken
+    ).concat(result?.diagnostics)
+    timingEnd("diagnostics")
+
+    const succeded =
+        !result.emitSkipped &&
+        result?.diagnostics.length === 0 &&
+        diagnostics.length == 0
+
+
+    if (!succeded) {
+        printDiagnostics(diagnostics)    
+    }
+    
+    workir.previousInputs = inputs;
+
+    return succeded
+}
+
+// Based on https://github.com/microsoft/TypeScript/blob/3fd8a6e44341f14681aa9d303dc380020ccb2147/src/executeCommandLine/executeCommandLine.ts#L465
+function emitOnce(args) {
+    const currentDirectory = ts.sys.getCurrentDirectory();
+    const reportDiagnostic = ts.createDiagnosticReporter({...ts.sys, write: worker.log});
+    const commandLine = ts.parseCommandLine(args)
+    const extendedConfigCache = new ts.Map();
+    const commandLineOptions = ts.convertToOptionsWithAbsolutePaths(commandLine.options, (fileName) => ts.getNormalizedAbsolutePath(fileName, currentDirectory));
+    const configParseResult = ts.parseConfigFileWithSystem(commandLine.options.project, commandLineOptions, extendedConfigCache, commandLine.watchOptions, ts.sys, reportDiagnostic);
+    const program = ts.createProgram({
+        options: configParseResult.options,
+        rootNames: configParseResult.fileNames,
+        projectReferences: configParseResult.projectReferences,
+        configFileParsingDiagnostics: configParseResult.errors
+    });
+    const result = program.emit();
+    const diagnostics = ts.getPreEmitDiagnostics(program).concat(result?.diagnostics)
+    const succeded =
+        !result.emitSkipped &&
+        result?.diagnostics.length === 0 &&
+        diagnostics.length == 0
+
+    if (!succeded) {
+        printDiagnostics(diagnostics)
+    }
+
+    return succeded
+}
+
+function main() {
+    if (worker.runAsWorker(process.argv)) {
+        worker.log(`Running ${MNEMONIC} as a Bazel worker`)
+        worker.runWorkerLoop(emit)
+    } else {
+        worker.log(`WARNING: Running ${MNEMONIC} as a standalone process`)
+        worker.log(`Your build might be misconfigured, try putting "build --strategy=${MNEMONIC}=worker" into your .bazelrc or add "supports_workers = False" attribute into this ts_project target.`)
+        worker.log(`Started a standalone process to perform this action but this might lead to some unexpected behavior with tsc due to being run non-sandboxed.`)
```

Comment:
Should this be one line up before the suggestion?

Reply:
Done. That's better.